source/vibe/core/process.d

```diff
@@ -0,0 +1,758 @@
+
+    return ProcessPipes(
+        Process(process.pid),
+        PipeOutputStream(cast(PipeFD)process.stdin),
+        PipeInputStream(cast(PipeFD)process.stdout),
+        PipeInputStream(cast(PipeFD)process.stderr)
+    );
+}
+
+/// ditto
+ProcessPipes pipeProcess(
+    scope string program,
+    Redirect redirect = Redirect.all,
+    const string[string] env = null,
+    Config config = Config.none,
+    scope NativePath workDir = NativePath.init)
+{
+    return pipeProcess(
+        [program],
+        redirect,
+        env,
+        config,
+        workDir
+    );
+}
+
+/// ditto
+ProcessPipes pipeShell(
+    scope string command,
+    Redirect redirect = Redirect.all,
+    const string[string] env = null,
+    Config config = Config.none,
+    scope NativePath workDir = NativePath.init,
+    scope NativePath shellPath = nativeShell)
+{
+    return pipeProcess(
+        shellCommand(command, nativeShell),
+        redirect,
+        env,
+        config,
+        workDir);
+}
+
+/**
+    Equivalent to `std.process.execute`.
+
+    Returns:
+        Tuple containing the exit status and process output.
+
+    See_Also: `spawnProcess`, `pipeProcess`
+*/
+auto execute(
+    scope string[] args,
+    const string[string] env = null,
+    Config config = Config.none,
+    size_t maxOutput = size_t.max,
+    scope NativePath workDir = NativePath.init)
+@blocking {
+    return executeImpl!pipeProcess(args, env, config, maxOutput, workDir);
+}
+
+/// ditto
+auto execute(
+    scope string program,
+    const string[string] env = null,
+    Config config = Config.none,
+    size_t maxOutput = size_t.max,
+    scope NativePath workDir = NativePath.init)
+@blocking @trusted {
+    return executeImpl!pipeProcess(program, env, config, maxOutput, workDir);
+}
+
+/// ditto
+auto executeShell(
+    scope string command,
+    const string[string] env = null,
+    Config config = Config.none,
+    size_t maxOutput = size_t.max,
+    scope NativePath workDir = null,
+    NativePath shellPath = nativeShell)
+@blocking {
+    return executeImpl!pipeShell(command, env, config, maxOutput, workDir, shellPath);
+}
+
+private auto executeImpl(alias spawn, Cmd, Args...)(
+    Cmd command,
+    const string[string] env,
+    Config config,
+    size_t maxOutput,
+    scope NativePath workDir,
+    Args args)
+@blocking {
+    Redirect redirect = Redirect.stdout;
+
+    auto processPipes = spawn(command, redirect, env, config, workDir, args);
+
+    auto stringOutput = processPipes.stdout.collectOutput(maxOutput);
+
+    return Tuple!(int, "status", string, "output")(processPipes.process.wait(), stringOutput);
+}
+
+/**
+    Collect the string output of a stream in a blocking fashion.
+
+    Params:
+        stream = The input stream to collect from.
+        nbytes = The maximum number of bytes to collect.
+
+    Returns:
+        The collected data from the stream as a string.
+*/
+string collectOutput(InputStream)(InputStream stream, size_t nbytes = size_t.max)
```

Comment:
In general, it probably makes sense to eventually move `readAll(UTF8)` to `vibe.core.stream` instead of `vibe.stream.operations`, since it is such basic functionality, similar to `pipe`.

Until then, since this is more or less the same, this one should have a doc comment `/// private` to remove it from the documentation (the current description can stay as a normal comment that precedes the doc comment).